zwad/aad.py

```diff
@@ -65,310 +45,209 @@ def __enter__(self):
-    ridxs_counts, region_extents = None, None
-    if len(ha) > 0:
-        ridxs_counts, region_extents = describe_instances(x, np.array(ha), model=model,
-                                                          opts=opts, interpretable=True)
-        logger.debug("selected region indexes and corresponding instance counts (among %d):\n%s" %
-                     (len(ha), str(list(ridxs_counts))))
-        logger.debug("region_extents: these are of the form [{feature_index: (feature range), ...}, ...]\n%s" %
-                     (str(region_extents)))
-    
-    return model, x_transformed, queried, ridxs_counts, region_extents
-
-
-def get_aad_option_list():
-    parser = aad_globals.get_aad_option_list()
-    parser.add_argument('--answers', metavar='FILENAME', action='append', help='answers.csv file for AnswersFileExpert')
-    parser.add_argument('--oid', metavar='FILENAME', action='append', help='Filepath to oid.dat', required=True)
-    parser.add_argument('--feature', metavar='FILENAME', action='append', help='Filepath to feature.dat', required=True)
-    parser.add_argument('--feature-names', metavar='FILENAME', help='Name of the file with feature names, one name per line')
-    parser.add_argument('--transform', action='store_true', help='Data transformation using nonlinear functions')
-    parser.add_argument('-n', '--non_interactive', action='store_true', help='Suppress InteractiveExpert')
-    parser.add_argument('-y', '--yes', action='store_true', help='Select \'yes\' for every choise')
-    parser.add_argument('--onnx_file', metavar='FILENAME', help='Dump trained model to ONNX format', required=False)
-    return parser
+        label = Label.ANOMALY if result else Label.REGULAR
 
+        return result, label
 
-class ZTFAadOpts(AadOpts):
-    def __init__(self, args):
-        super(ZTFAadOpts, self).__init__(args)
-        self.oid = args.oid
-        self.feature = args.feature
-        self.answers = args.answers
-        self.non_interactive = args.non_interactive
-        self.yes = args.yes
+    def _run_interactive(self, oid, feature, known):
+        budget = np.min([self._budget, feature.shape[0]])
+
+        for i in range(budget):
+            scores = self._anomaly_detector.score(feature)
+            scores_idx = np.argsort(scores)
+            a = next(filter(lambda idx: (idx not in known), scores_idx))
+            a_oid = oid[a]
+
+            result, label = self._evaluate(a_oid)
+
+            self._anomaly_detector.observe(np.atleast_1d(feature[a]), np.atleast_1d(label))
+            known[a] = label
+
+            yield (a_oid, result)
+
+    def _run_non_interactive(self, oid, feature):
+        budget = np.min([self._budget, feature.shape[0]])
+
+        scores = self._anomaly_detector.score(feature)
+        scores_idx = np.argsort(scores)
+
+        for i in range(budget):
+            a_oid = oid[scores_idx[i]]
+
+            yield (a_oid, True)
```

Comment:
Why is it always True?

Reply:
Because of the logic of how non-interactive mode should work. We are provided with prepared list of answers (and put this list as is into coinferest/aad), then we cannot ask expert for additional reasoning due to non-interactive mode. After the model is retrained using the prepared answers, the top of the list are our anomalies to be written to the output file.